content/posts/2021/reana-0.7.4.md

```diff
@@ -0,0 +1,134 @@
+---
+title: "REANA 0.7.4 is released"
+date: 2021-07-07T14:00:00+01:00
+tags:
+  - release
+  - client
+  - cluster
+---
+
+We are glad to announce the release of REANA 0.7.4, an update
+allowing REANA users to specify a memory limit for Kubernetes compute backend jobs.
+
+## What's new?
+
+### Custom memory limit for Kubernetes compute backend
+
+While executing a workflow there could be some cases when a complex workflow
+job with a high memory consumption exceeds the cluster memory limit. In this case
+the cluster will respond with out of memory (OOM) error and the job will be terminated (OOMKilled).
+To avoid such an error and to make an efficient use of the available resources,
+it is recommended to specify a memory limit per workflow step.
+If a memory limit for a specific step is increased, OOM error will not be thrown and
+the workflow will be executed as expected. Also, a memory limit could be decreased
+in order to increment the chances of jobs being scheduled earlier.
+
+REANA 0.7.4 enables to set a memory limit for a workflow jobs by specifying
+`kubernetes_memory_limit` in the specification of each workflow step.
+Please refer to
+[Kubernetes official documentation](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/#meaning-of-memory)
+for expected memory values.
+The specifications respectively can be configured for Serial, Yadage,
+and CWL workflow engines.
```

Comment:
The two paragraphs could be reformulated as follows:

REANA 0.7.4 enables users to set custom memory limits for their workflow jobs by specifying `kubernetes_memory_limit` in the resource clause specification of each workflow step. The section below will show examples for Serial, Yadage and CWL workflow engines.

Why is this important?  If you launch many workflows, the jobs "compete" against each other for available cluster resources, as it were. If you declare that a certain job is light regarding memory requirements, say 2 GiB, and the cluster has 40 GiB of free resources, the system would be able to schedule 20 such jobs in parallel. Setting the memory limit values will therefore provide a hint for the REANA system as to how many user workflows can be scheduled in order to arrive at your results faster.

Please note that if your job unexpectedly consumes more memory than declared, the job may be killed by the Kubernetes compute backend platform. This will be indicated by the out-of-memory killed message in the workflow logs (`OOMKilled`). In order to avoid these problems, please specify neither too low nor too high a limit, so that the jobs can be efficiently scheduled.

If you don't specify any custom memory limit value for some workflow step, then the REANA cluster will assume a certain default value that was set by the cluster administrator upon deployment. (Please see below.)